keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md

```diff
@@ -123,9 +123,9 @@ For alpha, the feature gates will be used for testing purposes. When enabled, te
 disabled by default.
 
 For beta, the feature gates will be on by default, meaning core components will disallow use of in-tree cloud providers. This will act as a warning for users to migrate to external components. Users may
-choose to continue using the in-tree provider by explicitly disabling the feature gates. Beta is targeted for v1.26 with the caveat that a majority of our CI signal jobs across providers should have converted to use CCM by then.
+choose to continue using the in-tree provider by explicitly disabling the feature gates. Beta is targeted for v1.28 with the caveat that a majority of our CI signal jobs across providers should have converted to use CCM by then.
 
-For GA, the feature gate will be enabled by default and locked. Users at this point MUST migrate to external components and use of the in-tree cloud providers will be disallowed. GA is targeted for v1.27. One release after GA, the in-tree cloud providers can be safely removed. 
+For GA, the feature gate will be enabled by default and locked. Users at this point MUST migrate to external components and use of the in-tree cloud providers will be disallowed. GA is targeted for v1.29. One release after GA, the in-tree cloud providers can be safely removed. 
```

Comment:
Sorry - can't comment on unchanged lines, so adding few comments here:

L204: I think it would make sense for this question to expand a bit on how exactly the migration should be done:
(a) to CCM [i guess it's primary https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/2436-controller-manager-leader-migration, so we may just want to link it]
(b) credential-provider
(c) network-proxy
(d) others?

In general, I would expect all separate components to be listed for completeness so that those who deploy their clusters can ensure the full setup
Also for those component, can the built-in and external versions can run at the same time?


L228: Were those tests created in the meantime? If not, can you please add an explicit Beta graduation criteria that they will get added and linked from this KEP?

Rollout, Monitoring and Dependecies sections should be filled-in - they are required for Beta.

Reply:
since this PR is just about bumping the versions, perhaps we should add that information as a followup?

my reasoning is that we have had this document as is for several releases and we do get questions frequently about when these changes will occur. i think there is value in updating the version info, even if some of the migration/testing information could use more input. just my thoughts.

Reply:
I agree @elmiko let's do the "how the migration" questions as a follow up

Reply:
updated GA version target

Reply:
@cheftako @andrewsykim as original authors of the KEP, please see notes from @wojtek-t above

Reply:
> since this PR is just about bumping the versions

It's not about bumping versions - it's about promoting the feature to Beta from Alpha.
The process of requiring prod-readiness beta questionaire was established for a reason and there are a number of questions required for Beta that are not answered.
I'm not going to approve it without that.

Reply:
@elmiko @bridgetkromhout will leave this upto you all to revisit when you feel ready.

Reply:
@wojtek-t understood. i think it will take us some time to fully enumerate the components and answer the questions.

we can certainly link the current docs for migration into this enhancement.

re: tests, we have some tests that live currently in the k/k e2e's and we are working on creating a more generic set of tests which could be used by all providers. this is still ongoing work.